kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/LogRepository.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2019 The Hyve
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.radarbase.connect.upload.converter
+
+import okio.BufferedSink
+import org.slf4j.Logger
+import java.time.Instant
+
+interface LogRepository {
+    fun recordLogger(logger: Logger, recordId: Long): RecordLogger
+    fun recordLogger(clazz: Class<*>, recordId: Long): RecordLogger
+
+    val recordIds: Set<Long>
+    fun extract(recordId: Long, reset: Boolean = false): Log?
+}
+
+interface RecordLogger {
```

Comment:
Maybe it is better to add documentation on what each of these interfaces mean? I am little lost when I see the method `recordLogger` and with an interface`RecordLogger`. when we call method recordLogger do we "record" a log, or we create an instance of `RecordLogger`?